src/coreclr/jit/lsra.cpp

```diff
@@ -10773,7 +10773,7 @@ void LinearScan::verifyFinalAllocation()
         {
             // Free Registers.
             // We could use the freeRegisters() method, but we'd have to carefully manage the active intervals.
-            for (regNumber reg = REG_FIRST; reg < ACTUAL_REG_COUNT; reg = REG_NEXT(reg))
+            for (regNumber reg = REG_FIRST; reg < ACTUAL_REG_COUNT && regsToFree != RBM_NONE; reg = REG_NEXT(reg))
```

Comment:
This new clause is a loop invariant, so maybe hoist it out...?
```suggestion
            if (regsToFree != RBM_NONE)
            {
               for (regNumber reg = REG_FIRST; reg < ACTUAL_REG_COUNT; reg = REG_NEXT(reg))
```

Or perhaps better yet (but maybe more involved) only iterate over the set bits of `regsToFree` using find first bit or similar tech (presumably `regsToFree` is mostly zeros so iterating over each bit is a little wasteful).

Also a few lines below the code seems odd, we should make sure we know which of the assignments is the right one.
```
            regsToFree = delayRegsToFree;
            regsToFree = RBM_NONE;
```

Reply:
Thanks for pointing it out about the `regsToFree = delayRegsToFree;` , I should have noticed it. But I did notice other problem where `delayRegsToFree` in this method is never assigned and is always `RBM_NONE` so the odd code doesn't do any harm. Probably, some day, I should check right way to introduce `delayRegsToFree`, but for now, I will fix the odd code as well as get rid of `delayRegsToFree` variable from this method.

Reply:
Also, `regsToFree` is never used and it was that way from initial commit (6 years back). So as a cleanup, I will also get rid of `regsFree` and investigate what is the right way as part of https://github.com/dotnet/runtime/issues/48837.